content/assignments/mini-libc/tests/test_stat.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+# SPDX-License-Identifier: BSD-3-Clause
+
+source graded_test.inc.sh
+
+exec_file=./io/stat
+
+# Create file.
+# Verify the following info: inode number, UID, GID, link count, number of blocks, size, last modify time.
+test_stat()
+{
+    if test ! -f "$exec_file"; then
+        echo "No such file $exec_file" 1>&2
+        exit 1
+    fi
+
+    nm "$exec_file" > /dev/null 2>&1
+    if test $? -ne 0; then
+        echo "No symbols in file" 1>&2
+        exit 1
+    fi
+
+    nm "$exec_file" | grep ' stat$' > /dev/null
+    if test $? -ne 0; then
+        echo "No stat symbol" 1>&2
+        exit 1
+    fi
+
+    touch ./existent_file         # create file
+
+    diff <(stat --format="%i %u %g %h %b %s %Y" "./existent_file") <("$exec_file")
```

Comment:
I would use `diff -u` here and in other places where `diff` is used. I believe the unified diff format (enabled by `-u`) is more common nowadays, for example in tools like `git`.